File: src/main/org/nlogo/prim/gui/_bench.java
package org.nlogo.prim.gui ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Syntax;
import org.nlogo.workspace.AbstractWorkspace;
import org.nlogo.workspace.Benchmarker;

public final strictfp class _bench
	extends org.nlogo.nvm.Command
{
	@Override public Syntax syntax()
	{
		return Syntax.commandSyntax
			( new int[] { Syntax.TYPE_NUMBER , Syntax.TYPE_NUMBER } ,
				"O---" ) ;
	}
	@Override public void perform( Context context ) throws LogoException
	{
		final int minTime = argEvalIntValue( context , 0 ) ;
		final int maxTime = argEvalIntValue( context , 1 ) ;
		new Thread( "__bench" ) {
			@Override public void run() {
				Benchmarker.benchmark( (AbstractWorkspace) workspace , minTime , maxTime ) ;
			} }.start() ;
		context.ip = next ;
	}
}

File: src/main/org/nlogo/prim/gui/_bench.scala
package org.nlogo.prim.gui

import org.nlogo.api.LogoException
import org.nlogo.nvm.{ Command, Context, Syntax }
import org.nlogo.workspace.{ AbstractWorkspace, Benchmarker }

class _bench extends Command {
  override def syntax =
    Syntax.commandSyntax(Array(Syntax.TYPE_NUMBER,
                               Syntax.TYPE_NUMBER),
                         "O---")
  override def perform(context: Context) {
    val minTime = argEvalIntValue(context, 0)
    val maxTime = argEvalIntValue(context, 1)
    new Thread("__bench") {
      override def run() {
        Benchmarker.benchmark(
          workspace.asInstanceOf[AbstractWorkspace], minTime, maxTime)
      }
    }.start()
    context.ip = next
  }
}
